internals/daemon/api_identities.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2024 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package daemon
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/canonical/pebble/internals/overlord/state"
+)
+
+func v1GetIdentities(c *Command, r *http.Request, _ *UserState) Response {
+	st := c.d.overlord.State()
+	st.Lock()
+	defer st.Unlock()
+
+	identities := st.Identities()
```

Comment:
probably should not just dump back all the identities straight from state for `UserAccess{}`. Maybe only `AdminAccess{}` can read every item, whereas, `UserAccess{}` can read their own, and the rest are partial?

Reply:
I was thinking UserAccess should be able to see the list too, kind of like as any user I can `cat /etc/passwd`. But no user, including admin, can read back any secrets. This keeps it simpler and (I think) less error-prone.

Note that that's why I've defined `Identity.MarshalJSON` (which uses the `apiIdentity` helper type), to ensure that when we add identity types with secrets, we won't add them there, and so they can't get exposed via the API at all. In contrast, when the state file is written to JSON on disk, `marshalledIdentity.MarshalJSON` takes care of that -- and this will include sensitive data or secrets.

Does that seem reasonable?